.github/workflows/test-and-publish.yml

```diff
@@ -0,0 +1,39 @@
+name: Test and Publish
+
+on:
+  push:
+
+jobs:
+  test-app:
+    name: Test App
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Run tests
+        run: docker-compose -f docker-compose.test.yml run --rm test ./run-tests.sh
+
+  build-and-publish:
+    name: Build and Publish
+    needs: test-app
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ vars.DOCKER_ORG }}/auth-proxy
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
```

Comment:
I am curious if this is needed, see [docs](https://github.com/docker/build-push-action):

<!DOCTYPE html>
context | String | Build's context is the set of files located in the specified PATH or URL(default Git context)
-- | -- | --


context	String	Build's context is the set of files located in the specified [PATH or URL](https://docs.docker.com/engine/reference/commandline/build/) (default [Git context](https://github.com/silinternational/auth-proxy/pull/63/files#git-context))